politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -1183,29 +1193,61 @@ func (p *commentsPlugin) cmdVotes(token []byte, payload string) (string, error)
 		return "", err
 	}
 
-	// Compile the comment vote digests for all votes that were cast
-	// by the specified user.
-	digests := make([][]byte, 0, 256)
+	// Compile the comment vote digests
+	var digests [][]byte
 	for _, cidx := range ridx.Comments {
-		voteIdxs, ok := cidx.Votes[v.UserID]
-		if !ok {
-			// User has not cast any votes for this comment
-			continue
+		var voteIdxs []voteIndex
+		if !filterByUserID {
+			// If no user ID filter is applied, collect all comment votes
+			for _, idxs := range cidx.Votes {
+				voteIdxs = append(voteIdxs, idxs...)
+			}
+		} else {
+			// If user ID filter is applied, collect only user's comment votes.
+			var ok bool
+			voteIdxs, ok = cidx.Votes[v.UserID]
+			if !ok {
+				// User has not cast any votes for this comment
+				continue
+			}
 		}
 
-		// User has cast votes on this comment
+		// Collect digests
 		for _, vidx := range voteIdxs {
 			digests = append(digests, vidx.Digest)
 		}
 	}
 
+	// If requested page exceeds the number of available pages, reutrn
+	// an emptry reply.
+	pageSize := p.votesPageSize
+	if len(digests) < int((page-1)*pageSize) {
+		return votesReply([]comments.CommentVote{})
+	}
+
 	// Lookup votes
 	votes, err := p.commentVotes(token, digests)
```

Comment:
Using this implemenation, what happens if there are 10,000 comment votes on a record and page 1 is requested?